WalletWasabi.Gui/Global.cs

```diff
@@ -488,39 +488,29 @@ public KeyManager LoadKeyManager(string walletFullPath)
 			return keyManager;
 		}
 
-		private void Coins_CollectionChanged(object sender, NotifyCollectionChangedEventArgs e)
+		private void CoinReceived(object sender, SmartCoin coin)
 		{
 			try
 			{
-				if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows) || UiConfig?.LurkingWifeMode is true)
+				if (coin.HdPubKey.IsInternal)
```

Comment:
we can combine this if with the if after it:
`if (coin.HdPubKey.IsInternal || RuntimeInformation.IsOSPlatform(OSPlatform.Windows) || UiConfig?.LurkingWifeMode is true)`

Reply:
It can be done but they are checking two different thinks, one is business logic and the other is presentation layer logic.